### PR TITLE
Show lifecycle status message base on Fine-tuning job exit code.

### PR DESCRIPTION
**Pull request by qiuosier**
Show lifecycle status details based on the job run exit code. The exit code mappings are obtained from the `exceptions.py`.

The `exceptions.py` is the same file from the fine-tuning package and it is used for raising the exceptions.
In the future, the fine-tuning package can import this file from ADS.

### CODE REVIEW

File: ads/aqua/training/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,459 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2024 Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""Contains exceptions classes for training job.
+
+Exit codes are defined in the exceptions.
```
Comment (VipulMascarenhas) [inline ads/aqua/training/exceptions.py:7]: there are 3 different ways within aqua to handle exit codes, this one is much better. we'll change the others to reflect like this in the future. :)

File: ads/aqua/training/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+
+
+class InvalidDataFormatErrorExit(TrainingExit):
+    """Error when training data format is invalid."""
+
+    code = 43
+
+    reason = (
+        "Unable to load training data. "
+        "Please provide training data in JSONL format with 'prompt' and 'completion'."
```
Comment (VipulMascarenhas) [inline ads/aqua/training/exceptions.py:136]: add "fields/columns" at the end to be consistent with MissingDataColumnErrorExit message.

File: ads/aqua/training/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+
+
+class InvalidDataValueErrorExit(TrainingExit):
+    """Error when training data contain non-string values."""
+
+    code = 45
+
+    reason = (
+        "Training data contain non-string values. "
+        "Please provide training data with only strings in JSONL"
```
Comment (VipulMascarenhas) [inline ads/aqua/training/exceptions.py:158]: maybe end with "... strings in JSONL format."

**Review by VipulMascarenhas — APPROVED** [review]
minor comments, overall looks good.

**Comment by mayoor** [comment]
Really good! Thanks!

File: ads/aqua/training/exceptions.py (hunk 1 above)
Comment (qiuosier) [reply ads/aqua/training/exceptions.py:7]: I didn't know there are multiple way 😅

File: ads/aqua/training/exceptions.py (hunk 2 above)
Comment (qiuosier) [reply ads/aqua/training/exceptions.py:136]: Added

**Pull request merged**